packages/ember-routing/lib/system/router.js

```diff
@@ -1285,7 +1287,7 @@ export function triggerEvent(handlerInfos, ignoreFailure, args) {
 
   let defaultHandler = defaultActionHandlers[name]
   if (defaultHandler) {
-    defaultHandler.apply(null, args);
+    defaultHandler.apply(this, [handlerInfos, ...args]);
```

Comment:
This makes the `defaultActionHandlers` above have the router as their `this` context (as opposed to `null`), and allows us to remove the reliance on `originRoute.router` to access the router for the implementations of `error` and `loading`.